### PR TITLE
wpa-supplicant: Add wpa_supplicant_conf from os-common

### PR DESCRIPTION
**Pull request by shruthi-ravi**
This script also generates a wpa_supplicant.conf file in
/etc/natinst/share. Therefore, remove the wpa_supplicant.conf file
installed by the OE recipe.

Built the recipe and verified the ipk contents for the presence of wpa_supplicant_conf and the absence of wpa_supplicant.conf.

@ni/rtos

### CODE REVIEW

**Review by amstewart — APPROVED** [review]

File: recipes-connectivity/wpa-supplicant/wpa-supplicant_2.%.bbappend [hunk 1]
```diff
@@ -5,11 +5,18 @@ SRC_URI =+ "file://wep-hexkeys-and-wpa-psk-hexkeys-fix.patch \
...
+            file://wpa_supplicant_conf;subdir=nilrt \
 "
 
 PACKAGECONFIG_append = " openssl"
 
 do_install_append () {
 	install -m 644 ${WORKDIR}/nilrt/dbus-wpa_supplicant.conf ${D}/${sysconfdir}/dbus-1/system.d
 	install -m 644 ${WORKDIR}/nilrt/*.service ${D}/${datadir}/dbus-1/system-services
+
+	# Remove the wpa_supplicant.conf file installed by the OE recipe. This will be replaced by
```
Comment (chaitu236) [inline recipes-connectivity/wpa-supplicant/wpa-supplicant_2.%.bbappend:17]: If we rename wpa_supplicant_conf to wpa_supplicant.conf, wouldn't bitbake automatically replace OE's version with this?

Comment (chaitu236) [reply recipes-connectivity/wpa-supplicant/wpa-supplicant_2.%.bbappend:17]: nvm. I thought you were installing the file as wpa_supplicant.conf.

**Comment by chaitu236** [comment]
lgtm

File: recipes-connectivity/wpa-supplicant/wpa-supplicant_2.%.bbappend (hunk 1 above)
Comment (shruthi-ravi) [reply recipes-connectivity/wpa-supplicant/wpa-supplicant_2.%.bbappend:17]: wpa_supplicant_conf is a shell script. Lines 7-12 in that script generate the .conf file if it doesn't already exist. So it is not a 1-1 replacement.

Comment (chaitu236) [reply recipes-connectivity/wpa-supplicant/wpa-supplicant_2.%.bbappend:17]: Yes, hadn't read the description properly

File: recipes-connectivity/wpa-supplicant/files/wpa_supplicant_conf [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+# Copyright (c) 2012-2014 National Instruments
+# All Rights Reserved
+
+WPA_SUP_CONF=/etc/natinst/share/wpa_supplicant.conf
+
+# If the supplicant config file is missing, create it
+if [ ! -e $WPA_SUP_CONF ] && [ -x /usr/sbin/wpa_supplicant ]; then
+    cat <<TXT >/etc/natinst/share/wpa_supplicant.conf
```
Comment (jpautler) [inline recipes-connectivity/wpa-supplicant/files/wpa_supplicant_conf:9]: Could the path on this line also be replaced with $WPA_SUP_CONF?

**Pull request merged**